Move saved-RES tab to save space, improve UX

githue opened the pull request:
Created some inline tabs to save space in the main tab list. The URL remains the same so old links to it will still work. Cleaned up the formatting too.

[screenshot below]

Not sure if I'm using history() correctly since I haven't used it before. Not sure if data param is necessary.

erikdesjardins commented:
The NER indicator doesn't get hidden when you change tabs; otherwise it looks good.
![image](https://cloud.githubusercontent.com/assets/7673145/9983407/71d19dc4-5fc8-11e5-8f7b-693ddda058b9.png)

It appears that the `history.pushState` data can be null or undefined, but it doesn't really matter since we don't use it.

githue commented:
![saved-comments-v2](https://cloud.githubusercontent.com/assets/7245595/12508580/86433c7c-c148-11e5-914d-3f3f76b2d2b4.png)

Updated the UI and refactored the code.

* Now using templates.html
* Makes the whole header of each comment a link to the comment.
* Gets the subreddit name and puts it in the header (is the regex ok?).
* When you unsave a comment, it keeps the comment until you reload the page, this allows you to recover from a miss-click.
* Hovering over "X days ago" shows a tooltip with the exact date.

erikdesjardins commented on lib/core/templates.html:
```diff
@@ -573,5 +573,50 @@ <h3 class="RESHoverTitle" data-hover-element="0"></h3>
+			</div>
+		</div>
+
+		<div id="saveComments-comment">
```
Would be cool if this was all in the same template, in a `{{#comments}}` list.
Then you could eliminate `$wrapper.find('.res-module-description').hide();`, and wrap the description in `{{^comments}}{{/comments}}`, which will only be displayed if `comments` is falsy or an empty list.

erikdesjardins commented on lib/modules/saveComments.js:
```diff
@@ -167,122 +159,118 @@ addModule('saveComments', function(module, moduleID) {
+		}
+
+		if (!modules['keyboardNav'].isEnabled()) {
+			$wrapper.find('.res-module-tip').hide();
```
This can be done in the template with `{{#keyboardNavEnabled}}...{{/keyboardNavEnabled}}`.

That is: if `foo` isn't an array, then `{{#foo}}bar{{/foo}}{{^foo}}baz{{/foo}}` acts like `foo ? 'bar' : 'baz'`.
